sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1593,7 +1596,9 @@ class Dataset[T] private[sql](
   def groupBy(col1: String, cols: String*): RelationalGroupedDataset = {
     val colNames: Seq[String] = col1 +: cols
     RelationalGroupedDataset(
-      toDF(), colNames.map(colName => resolve(colName)), RelationalGroupedDataset.GroupByType)
+      toDF(),
+      colNames.distinct.map(colName => resolve(colName)),
```

Comment:
This will cause a behavior change.

Reply:
Yes, this ticket is only about making this behavior change. So the question really is if the team think that it is the expected behavior. I personally find it helps eliminate bugs in our ETL. I don't think anyone need to have duplicated columns in their grouped data set by intention.